llap-server/src/java/org/apache/hadoop/hive/llap/security/LlapUgiFactoryFactory.java

```diff
@@ -15,42 +15,115 @@
 package org.apache.hadoop.hive.llap.security;
 
 import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
 
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.hive.common.UgiFactory;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.conf.HiveConf.ConfVars;
 import org.apache.hadoop.hive.llap.LlapUtil;
 import org.apache.hadoop.hive.shims.HadoopShims;
 import org.apache.hadoop.hive.shims.ShimLoader;
+import org.apache.hadoop.security.Credentials;
 import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** No Java application is complete until it has a FactoryFactory. */
 public class LlapUgiFactoryFactory {
```

Comment:
name is crazy, could we rename to say `LlapUgiHelper`?

Reply:
this craziness is already part of hive come, I'm fine with changing it to any of your suggestions

Reply:
````
LlapUgiManager {
   private ConcurrentHashMap<Identifier, UserGroupInformation > fsTaskUgiMap;
   private UgiFactory factory;
   
  public LlapUgiManager getInstance(Configuration conf) {
    return new LlapUgiManager(conf);
  }
  private LlapUgiManager(Configuration conf){
      try {
        fsUgiFactory = LlapUgiFactoryFactory.createFsUgiFactory(daemonConf);
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
   }
}
````